js/src/scatter.ts

```diff
@@ -415,33 +420,40 @@ class ScatterView extends widgets.WidgetView {
             this.line_material_rgb.linewidth = this.line_material.linewidth = this.model.get("line_material").obj.linewidth;
         }
 
+        // TODO: lighting_model for lines could be different
+        this.lighting_model = materialToLightingModel(this.material)
         const shader_snippets = this.model.get('shader_snippets');
         const snippet_defines = {};
         for (const key of Object.keys(shader_snippets)) {
             snippet_defines["SHADER_SNIPPET_" + key.toUpperCase()] = shader_snippets[key];
         }
-
-        this.material.defines = {USE_COLORMAP: this.model.get("color_scale") !== null, ...this.scale_defines, ...snippet_defines};
+        this.material.defines = {USE_COLOR: true, USE_COLORMAP: this.model.get("color_scale") !== null, ...this.scale_defines, ...snippet_defines};
+        this.material.defines[`AS_${this.lighting_model}`] = true;
         this.material.extensions = {derivatives: true};
-        this.material_rgb.defines = {USE_RGB: true, ...this.scale_defines, ...snippet_defines};
+        this.material_rgb.defines = {AS_COORDINATE: true, USE_COLOR: true, ...this.scale_defines, ...snippet_defines};
         this.material_rgb.extensions = {derivatives: true};
-        this.line_material.defines = {AS_LINE: true, ...this.scale_defines,  ...snippet_defines};
-        this.line_material_rgb.defines = {USE_RGB: true, AS_LINE: true};
+        this.line_material.defines = {IS_LINE: true, USE_COLOR: true, ...this.scale_defines,  ...snippet_defines};
+        this.line_material.defines[`AS_${this.lighting_model}`] = true;
+        this.line_material_rgb.defines = {AS_COORDINATE: true, IS_LINE: true, USE_COLOR: true};
         // locally and the visible with this object's visible trait
         this.material.visible = this.material.visible && this.model.get("visible");
         this.material_rgb.visible = this.material.visible && this.model.get("visible");
         this.line_material.visible = this.line_material.visible && this.model.get("visible");
         this.line_material_rgb.visible = this.line_material.visible && this.model.get("visible");
+
+
+
         this.materials.forEach((material) => {
-            material.vertexShader = (require("raw-loader!../glsl/scatter-vertex.glsl") as any).default;
-            material.fragmentShader = (require("raw-loader!../glsl/scatter-fragment.glsl") as any).default;
-            material.uniforms = {...material.uniforms, ...this.uniforms};
-            material.depthWrite = true;
-            material.transparant = true;
-            material.depthTest = true;
+            material.onBeforeCompile = (shader) => {
+                shader.vertexShader = vertexShader;
+                shader.fragmentShader = fragmentShader;
+                shader.uniforms = {...shader.uniforms, ...this.uniforms};
+                patchShader(shader);
+            };
             material.needsUpdate = true;
-            patchMaterial(material);
+            material.lights = true;
```

Comment:
```suggestion
```
No need for this if we set it in the ShaderMaterial constructor from the Python side (Phong, Physical etc don't seem to need this)